internal/restyutil/restyutil.go

```diff
@@ -0,0 +1,36 @@
+package restyutil
+
+import (
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/config"
+	goclienterr "github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+)
+
+type Authenticator interface {
+	Authenticate(r *resty.Request) error
+}
+
+func NewHTTPClient(cfg config.Config, auth Authenticator) *resty.Client {
+	return resty.New().
+		SetTransport(cfg.Transport).
+		SetBaseURL(cfg.Addr).
+		SetTimeout(cfg.Timeout).
+		OnBeforeRequest(func(_ *resty.Client, r *resty.Request) error {
+			return auth.Authenticate(r)
+		}).
+		SetError(&models.SPVError{}).
+		OnAfterResponse(func(_ *resty.Client, r *resty.Response) error {
+			if r.IsSuccess() {
+				return nil
+			}
+
+			if spvError, ok := r.Error().(*models.SPVError); ok && len(spvError.Code) > 0 {
```

Comment:
Why is there a check for a length of the code?

Reply:
Good question, the obvious answer is because of the string type. However, if API will respond with the SPVError format and by some reason the code will not be set (i.e empty string) then instead of returning the spvError we'll return something totally different. 

However, I created a task that covers the code in the on after response middleware: https://bitcoinassociation.atlassian.net/browse/SPV-1217  😄